llama_index/multi_modal_llms/base.py

```diff
@@ -0,0 +1,137 @@
+from abc import abstractmethod
+from enum import Enum
+from typing import Any, AsyncGenerator, Generator, Optional, Sequence
+
+from llama_index.bridge.pydantic import BaseModel, Field
+from llama_index.constants import (
+    DEFAULT_CONTEXT_WINDOW,
+    DEFAULT_NUM_INPUT_FILES,
+    DEFAULT_NUM_OUTPUTS,
+)
+from llama_index.schema import BaseComponent, ImageDocument
+
+
+class MessageRole(str, Enum):
+    """Message role."""
+
+    SYSTEM = "system"
+    USER = "user"
+    ASSISTANT = "assistant"
+    FUNCTION = "function"
+
+
+# ===== Generic Model Output - Completion =====
+class MultiModalCompletionResponse(BaseModel):
+    """
+    Completion response.
+
+    Fields:
+        text: Text content of the response if not streaming, or if streaming,
+            the current extent of streamed text.
+        additional_kwargs: Additional information on the response(i.e. token
+            counts, function calling information).
+        raw: Optional raw JSON that was parsed to populate text, if relevant.
+        delta: New text that just streamed in (only relevant when streaming).
+    """
+
+    text: str
+    additional_kwargs: dict = Field(default_factory=dict)
+    raw: Optional[dict] = None
+    delta: Optional[str] = None
+
+    def __str__(self) -> str:
+        return self.text
+
+
+MultiModalCompletionResponseGen = Generator[MultiModalCompletionResponse, None, None]
+MultiModalCompletionResponseAsyncGen = AsyncGenerator[
+    MultiModalCompletionResponse, None
+]
+
+
+class MultiModalLLMMetadata(BaseModel):
+    context_window: int = Field(
+        default=DEFAULT_CONTEXT_WINDOW,
+        description=(
+            "Total number of tokens the model can be input when generating a response."
+        ),
+    )
+    num_output: int = Field(
+        default=DEFAULT_NUM_OUTPUTS,
+        description="Number of tokens the model can output when generating a response.",
+    )
+    num_input_files: int = Field(
+        default=DEFAULT_NUM_INPUT_FILES,
+        description="Number of input files the model can take when generating a response.",
+    )
+    is_chat_model: bool = Field(
```

Comment:
is this used anywhere? looks like the multimodal LMM class doesn't have a chat interface yet?